Framework/API/src/AnalysisDataService.cpp

```diff
@@ -166,6 +167,22 @@ void AnalysisDataServiceImpl::remove(const std::string &name) {
   if (ws) {
     ws->setName("");
   }
+  return ws;
+}
+
+/**
+ * Make a new WorkspaceGroup and place the named member inside of it if the member is not already a WorkspaceGroup.
+ * @param name The name of a workspace to place into a new WorkspaceGroup.
+ * @return The created or existing WorkspaceGroup
+ */
+WorkspaceGroup_sptr AnalysisDataServiceImpl::makeGroup(std::string const &name) {
+  if (auto const workspace = retrieveWS<WorkspaceGroup>(name)) {
+    return workspace;
+  }
+  auto wsGroup = std::make_shared<Mantid::API::WorkspaceGroup>();
+  wsGroup->addWorkspace(remove(name));
```

Comment:
Sorry if I missed something. But I don't understand well why is necessary to call the remove func, and to change remove to retrieve the shared pointer?

Reply:
I think I returned the shared pointer from `remove` because it is convenient to remove an item from a data structure and get it at the same time, a bit like the `pop` method for a python dictionary. I'm open to other suggestions

Reply:
So, the workspace is removed from the ads, but not really deleted? Where does it live?

Reply:
Yes, so the workspace remains in memory but is removed from the ADS. You can think of the ADS as a container such as a dictionary. When you pop an item from the dictionary it is removed from the dictionary, but still exists in memory if it is assigned to another variable.